spec/after_commit_action_spec.rb

```diff
@@ -18,4 +19,21 @@
     t.array.should include('before_update')
     t.array.should include('after_update')
   end
+
+  context 'when block is executed in another model transaction' do
+    let(:tester)        { Tester.create count: 0 }
+    let(:another_model) { AnotherModel.new tester: tester }
+
+    subject { another_model.save! }
+
+    it { expect { subject }.to change { tester.reload.count }.from(0).to 1 }
+  end
+
+  context 'when there is no transaction' do
```

Comment:
it runs without a transaction, because increment_counter is just a custom method. I agree, that it is misleading, rails also adds the same method, better to call it in another way.